mesonbuild/backend/backends.py

```diff
@@ -737,37 +741,40 @@ def get_external_rpath_dirs(self, target: build.BuildTarget) -> T.Set[str]:
+        paths: OrderedSet[str] = OrderedSet()
         for dep in target.external_deps:
             if not isinstance(dep, (dependencies.ExternalLibrary, dependencies.PkgConfigDependency)):
                 continue
-            la = dep.link_args
-            if len(la) != 1 or not os.path.isabs(la[0]):
-                continue
-            # The only link argument is an absolute path to a library file.
-            libpath = la[0]
-            libdir = os.path.dirname(libpath)
-            if exclude_system and self._libdir_is_system(libdir, target.compilers, self.environment):
-                # No point in adding system paths.
-                continue
-            # Don't remove rpaths specified in LDFLAGS.
-            if libdir in self.get_external_rpath_dirs(target):
-                continue
-            # Windows doesn't support rpaths, but we use this function to
-            # emulate rpaths by setting PATH, so also accept DLLs here
-            if os.path.splitext(libpath)[1] not in ['.dll', '.lib', '.so', '.dylib']:
-                continue
-            if libdir.startswith(self.environment.get_source_dir()):
-                rel_to_src = libdir[len(self.environment.get_source_dir()) + 1:]
-                assert not os.path.isabs(rel_to_src), f'rel_to_src: {rel_to_src} is absolute'
-                paths.append(os.path.join(self.build_to_src, rel_to_src))
-            else:
-                paths.append(libdir)
+            for libpath in dep.link_args:
+                # For all link args that are absolute paths to a library file, add RPATH args
+                if not os.path.isabs(libpath):
+                    continue
+                libdir = os.path.dirname(libpath)
+                if exclude_system and self._libdir_is_system(libdir, target.compilers, self.environment):
+                    # No point in adding system paths.
+                    continue
+                # Don't remove rpaths specified in LDFLAGS.
+                if libdir in self.get_external_rpath_dirs(target):
+                    continue
+                # Windows doesn't support rpaths, but we use this function to
+                # emulate rpaths by setting PATH, so also accept DLLs here
+                if os.path.splitext(libpath)[1] not in ['.dll', '.lib', '.so', '.dylib']:
+                    continue
+                if libdir.startswith(self.environment.get_source_dir()):
+                    rel_to_src = libdir[len(self.environment.get_source_dir()) + 1:]
+                    assert not os.path.isabs(rel_to_src), f'rel_to_src: {rel_to_src} is absolute'
+                    paths.add(os.path.join(self.build_to_src, rel_to_src))
+                else:
+                    paths.add(libdir)
+            # Don't remove rpaths specified by the dependency
+            for d in self.get_rpath_dirs_from_link_args(dep.link_args):
+                if d in paths:
+                    paths.remove(d)
```

Comment:
```suggestion
            paths.update_difference(self.get_rpath_dirs_from_link_args(dep.link_args))
```

I think?

Reply:
`difference_update()` does indeed do what I want it to, thanks!

Reply:
Hah, it wasn't implemented for our implementation of OrderedSet, so I added it.